cpp/src/io/csv/reader_impl.cu

```diff
@@ -346,35 +340,43 @@ table_with_metadata reader::impl::read(size_t range_offset,
   std::vector<column_buffer> out_buffers;
   for (int col = 0, active_col = 0; col < num_actual_cols; ++col) {
     if (h_column_flags[col] & column_parse::enabled) {
+      // Replace EMPTY dtype with STRING
+      if (column_types[active_col].id() == type_id::EMPTY) {
+        column_types[active_col] = data_type{STRING};
+      }
       out_buffers.emplace_back(column_types[active_col], num_records, true, stream, mr_);
       metadata.column_names.emplace_back(col_names[col]);
       active_col++;
     }
   }
 
-  if (num_records != 0) { decode_data(column_types, out_buffers, stream); }
-
-  for (size_t i = 0; i < column_types.size(); ++i) {
-    out_columns.emplace_back(make_column(column_types[i], num_records, out_buffers[i]));
-  }
-
-  // TODO: String columns need to be reworked to actually copy characters in
-  // kernel to allow skipping quotation characters
-  /*for (auto &column : columns) {
-    column.finalize();
-
-    // PANDAS' default behavior of enabling doublequote for two consecutive
-    // quotechars in quoted fields results in reduction to a single quotechar
-    if (column->dtype == GDF_STRING &&
-        (opts.quotechar != '\0' && opts.doublequote == true)) {
-      const std::string quotechar(1, opts.quotechar);
-      const std::string dblquotechar(2, opts.quotechar);
-      auto str_data = static_cast<NVStrings *>(column->data);
-      column->data = str_data->replace(dblquotechar.c_str(), quotechar.c_str());
-      NVStrings::destroy(str_data);
+  if (num_records != 0) {
+    decode_data(column_types, out_buffers, stream);
+
+    for (size_t i = 0; i < column_types.size(); ++i) {
+      if (column_types[i].id() == type_id::STRING && opts.quotechar != '\0' &&
+          opts.doublequote == true) {
+        // PANDAS' default behavior of enabling doublequote for two consecutive
+        // quotechars in quoted fields results in reduction to a single quotechar
+        // TODO: Would be much more efficient to perform this operation in-place
+        // during the conversion stage
+        const std::string quotechar(1, opts.quotechar);
+        const std::string dblquotechar(2, opts.quotechar);
+        std::unique_ptr<column> col =
+          make_strings_column(out_buffers[i]._strings, stream /*, mr_*/);
```

Comment:
I think you can delete the commented out code here -- it is indeed correct to not pass the user-specified mr (`mr_`) so the default resource will be used.

```suggestion
          make_strings_column(out_buffers[i]._strings, stream);
```

Reply:
Ok with me. Was worried that replace could return the input if no characters were replaced.